working dir/widgets/playlistView/PlaylistView.kt

```diff
@@ -0,0 +1,242 @@
+                            lastSavePlaylistLocation ?: PlayerConfiguration.lastSavePlaylistLocation,
+                            "Playlist",
+                            widget.windowOrActive.orNull()?.stage ?: APP.windowManager.createStageOwner(),
+                            FileChooser.ExtensionFilter("m3u8", "m3u8")
+                    ).ifOk { file ->
+                        lastSavePlaylistLocation = file.parentDirOrRoot
+                        PlayerConfiguration.lastSavePlaylistLocation = file.parentDirOrRoot
+                        runFut { writePlaylist(table.selectedOrAllItemsCopy, file.name, file.parentDirOrRoot) }
+                    }
+                }
+        )
+        table.menuOrder.items addToStart Menu("Order by").apply {
+            items += Field.FIELDS.map { f -> menuItem(f.toStringEnum()) { table.sortBy(f) } }
+        }
+        onClose += table.selectionModel.selectedItemProperty() attach {
+            if (!table.movingItems)
+                outSelected.value = it
+        }
+        onClose += table::dispose
+
+        onScroll = EventHandler { it.consume() }
+
+    }
+
+    override fun getFields(): Collection<Config<Any>> {
+        widget.properties["columns"] = table.columnState.toString()
+        return super.getFields()
+    }
+
+    override fun refresh() {
+        runOn(once) {
+            table.columnState = widget.properties.getS("columns")?.net { TableColumnInfo.fromString(it) } ?: table.defaultColumnInfo
+        }
+        filter_for_playback.applyValue()
+    }
+
+    override fun getPlaylist() = playlist
+
+    private fun filterToggle(): Unit = filter_for_playback.setCycledNapplyValue()
+
+    private fun computeInitialPlaylist(id: UUID) = null
+            ?: PlaylistManager.playlists[id]
+            ?: getUnusedPlaylist(id).apply {
+                PlaylistManager.playlists.put(this)
+            }
+
+    private fun getUnusedPlaylist(id: UUID): Playlist {
+        val danglingPlaylists = ArrayList(PlaylistManager.playlists)
+        APP.widgetManager.widgets.findAll(OPEN).asSequence()
+                .filter { it.info.hasFeature(PlaylistFeature::class.java) }
+                .mapNotNull { (it.controller as PlaylistFeature?)?.playlist }
+                .forEach { danglingPlaylists.removeIf { playlist -> playlist.id==it.id } }
+
+        val danglingPlaylist: Playlist? = null
+                ?: danglingPlaylists.find { it.id==PlaylistManager.active }
+                ?: danglingPlaylists.firstOrNull()
+
+        if (danglingPlaylist!=null) {
+            PlaylistManager.playlists.remove(danglingPlaylist)
+
+            if (danglingPlaylist.id==PlaylistManager.active)
+                PlaylistManager.active = id
+            Util.setField(Playlist::class.java, danglingPlaylist, Playlist::id.name, id) // TODO: fix
+        }
+
+        return danglingPlaylist ?: Playlist(id)
+    }
+
+    companion object {
+        infix fun <T> MutableList<T>.addToStart(item: T) = add(0, item)
```

Comment:
Hmmmm, I suppose so. Ok.

Reply:
I'm not putting this to util as the add(0, iem) is good enough - nobody needs this function. It exists for its single use in the PlaylistView to improve the parenthesis hell. If this was in util, it would certainly not be infix fun.